parsec/core/backend_connection/invited.py

```diff
@@ -2,19 +2,25 @@
 from __future__ import annotations
 
 import trio
-from contextlib import asynccontextmanager
-from typing import Optional, AsyncGenerator
+from contextlib import _AsyncGeneratorContextManager, asynccontextmanager
+from typing import Any, Callable, Optional, AsyncGenerator, TypeVar, Union
 
 from parsec.api.protocol import INVITED_CMDS
-from parsec.core.types import BackendInvitationAddr
+from parsec.core.types import BackendAddrType, BackendInvitationAddr
 from parsec.core.backend_connection import cmds
 from parsec.core.backend_connection.transport import connect_as_invited
 from parsec.core.backend_connection.exceptions import BackendNotAvailable
-from parsec.core.backend_connection.expose_cmds import expose_cmds_with_retrier
+from parsec.core.backend_connection.expose_cmds import Transport, expose_cmds_with_retrier
+
+T = TypeVar("T")
 
 
 class BackendInvitedCmds:
-    def __init__(self, addr, acquire_transport):
+    def __init__(
+        self,
+        addr: BackendAddrType,
+        acquire_transport: Callable[..., _AsyncGeneratorContextManager[T]],
```

Comment:
Maybe use [AbstractAsyncContextManager](https://docs.python.org/3/library/contextlib.html#contextlib.AbstractAsyncContextManager)?
Also see: https://docs.python.org/3/library/typing.html#typing.AsyncContextManager

Reply:
I've used the `AcquireTransport` protocol I've defined